packages/presentation/lib/screen/login/login_bloc.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:domain/model/user/user_model.dart';
+import 'package:flutter/material.dart';
+import 'package:presentation/base/bloc.dart';
+import 'package:domain/usecase/check_user_usecase.dart';
+import 'package:domain/usecase/google_auth_usecase.dart';
+import 'package:domain/usecase/facebook_auth_usecase.dart';
+import 'package:domain/usecase/analytics_usecase.dart';
+import '../profile/profile_screen.dart';
+import 'login_data.dart';
+import 'login_screen.dart';
+
+abstract class LoginBloc extends Bloc<LoginScreenArguments, LoginData> {
+  factory LoginBloc(
+    CheckUserUseCase checkUserUseCase,
+    GoogleAuthUseCase googleAuthUseCase,
+    FacebookAuthUseCase facebookAuthUseCase,
+      LogAnalyticsEventUseCase logAnalyticsEventUseCase,
+  ) =>
+      _LoginBloc(
+        checkUserUseCase,
+        googleAuthUseCase,
+        facebookAuthUseCase,
+        logAnalyticsEventUseCase,
+      );
+
+  TextEditingController get emailController;
+
+  TextEditingController get passwordController;
+
+  Future<void> onLogin();
+
+  Future<void> authByGoogle();
+
+  Future<void> authByFacebook();
+}
+
+class _LoginBloc extends BlocImpl<LoginScreenArguments, LoginData>
+    implements LoginBloc {
+  LoginData _screenData = LoginData(
+    '',
+    '',
+  );
+  final _emailController = TextEditingController();
+  final _passwordController = TextEditingController();
+  final CheckUserUseCase _checkUserUseCase;
+  final GoogleAuthUseCase _googleAuthUseCase;
+  final FacebookAuthUseCase _facebookAuthUseCase;
+  final LogAnalyticsEventUseCase _logAnalyticsEventUseCase;
+
+  _LoginBloc(
+    this._checkUserUseCase,
+    this._googleAuthUseCase,
+    this._facebookAuthUseCase,
+    this._logAnalyticsEventUseCase,
+  );
+
+  @override
+  void initArgs(LoginScreenArguments arguments) {
+    super.initArgs(arguments);
+    _screenData = LoginData(
+      '',
+      '',
+    );
+    _updateData();
+  }
+
+  _updateData({bool? isLoading}) {
+    handleData(
+      data: _screenData,
+      isLoading: isLoading,
+    );
+  }
+
+  @override
+  Future<void> onLogin() async {
+    await _logAnalyticsEventUseCase('auth_by_login');
+    final String email = _emailController.text.trim();
+    final String password = _passwordController.text.trim();
+    final UserModel user = UserModel(email, password);
+    final bool isSuccess = await _checkUserUseCase(user);
+    pushSuccessScreen(isSuccess);
+  }
+
+  @override
+  Future<void> authByGoogle() async {
+    await _logAnalyticsEventUseCase('auth_by_google');
+    final UserModel? user = await _googleAuthUseCase();
+    _emailController.text = user?.email ?? '';
+    _passwordController.text = user?.password ?? '';
+    UserModel userModel = UserModel(
+      user?.email ?? '',
+      user?.password ?? '',
+    );
+    final bool isSuccess = await _checkUserUseCase(userModel);
+    pushSuccessScreen(isSuccess);
```

Comment:
remove the success condition from push screen. wrap the method into the condition